src/linear/updater_gpu_coordinate.cu

```diff
@@ -167,33 +167,28 @@ class GPUCoordinateUpdater : public LinearUpdater {
                       const gbm::GBLinearModelParam &model_param) {
     if (!shards_.empty()) return;
 
-    dist_ = GPUDistribution::Block(GPUSet::All(learner_param_->gpu_id, learner_param_->n_gpus,
-                                               p_fmat->Info().num_row_));
-    auto devices = dist_.Devices();
+    device_ = learner_param_->gpu_id;
 
-    size_t n_devices = static_cast<size_t>(devices.Size());
     size_t row_begin = 0;
```

Comment:
i think this entire block can be removed and `0, num_row` directly passed to `DeviceShard` constructor; even better if `DeviceShard` does not use row indices any more (and uses num_row)

Reply:
Done.